### PR TITLE
zkEVM: Add aggregator doc

### PR DESCRIPTION
**Pull request by EmpieichO**
zkEVM: Adding a new doc "aggregator-proofs.md" ... as per issue https://polygon.atlassian.net/browse/DOCS-296

### CODE REVIEW

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+Batch execution is carried out for any input data, resulting in the generation of a proof that confirms a state change for valid input data, or no state change for invalid input data, as depicted in the figure below.
+
+This strategy guarantees robust validation of execution outcomes, and provides a reliable mechanism to handle potential, malicious behavior.
+
+![Figure: Correct vs. wrong state change](../../../img/zkEVM/ag-correct-vs-wrong-state-change.png)
+
+
+### Invalid transactions
+
+Let us describe some errors in transactions, that cause the state to remain unchanged, as shown in the figure below.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
Let us describe some errors in transactions that cause the state to remain unchanged, as shown in the figure below.
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+- Having a stack that is too large.
+
+- Encountering a revert call in the code.
+
+This is a common scenario in EVM processing.
+
+#### Invalid intrinsic transaction
+
+An invalid intrinsic transaction is a transaction that cannot be processed and has no impact on the current state.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
An invalid intrinsic transaction is a transaction that cannot be processed, and thus has no impact on the current state.
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+![Figure: Invalid or reverted txs](../../../img/zkEVM/ag-invalid-or-reverted-txs.png)
+
+### Invalid batches
+
+Some of the errors that occur at the batch level can cause the state to remain unchanged.
+
+As depicted in the figure below, such errors might occur due to invalid data, or exhaustion of prover resources.
+
+- Invalid Data: Failure to decode RLP-encoded transactions, that is having garbage input.
+- Prover resource exhaustion: The zkEVM manages resources in terms of row counters in the Stack, also known as _zkCounters_. Processing a batch may fail due running out of zkCounters (OOC).
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
- Prover resource exhaustion: The zkEVM prover manages resources in terms of row counters in the Stack, also known as _zkCounters_. Processing a batch may fail due running out of zkCounters (OOC).
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+The main use case is to allow a user to send bridge transactions to withdraw assets from L2 without the risk of censorship, which would otherwise make fund withdrawals impossible.
+
+Since every user who sends L2 batch data is, by default, 'untrusted,' the system must ensure that anything sent by any user can be verified.
+
+The _forced batches_ mechanism is elaborated on in the [Malfunction resistance subsection](../../../zkEVM/architecture/protocol/malfunction-resistance/sequencer-resistance.md).
+
+## zkCounters
+
+Counters are used, during the execution of a batch, to monitor the row count in each state machine, including the main state machine.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
Counters are used during the execution of a batch to monitor the row count in each state machine, including the main state machine.
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+
+Management of these counters is incorporated within the computation process.
+
+By so doing, if a computation exhausts its assigned resources while generating the proof, then the system is able to prove that the batch does not cause a change in the state.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
The management of these counters is incorporated within the computation process. This way, if a computation exhausts its assigned resources while generating the proof, the system can verify and prove that the batch does not cause a state change.
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+
+The figure below depicts a simplified scenario of the OOC error for a [Keccak state machine](../zkprover/hashing-state-machines/keccakf-sm.md), where its arithmetization permits only up to four Keccak operations before exhausting the available rows.
+
+So, an OOC error occurs if a transaction invokes five or more Keccak operations.
+
+![Figure: Executor ... possible keccaks](../../../img/zkEVM/ag-executor-possible-keccaks.png)
+
+### VADCOPs
+
+Although the current version of the proving system has this limitation in the backend (forcing execution traces of all state machines to have the same amount of rows), it is expected to be removed when a proving technique called _variable degree composite proofs_ (VADCOPs) is implemented.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
Although the current version of the proving system has this limitation in the backend (forcing execution traces of all state machines to have the same amount of rows), it is expected to be resolved with the implementation of a proving technique called _variable degree composite proofs_ (VADCOPs).
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+See the figure below for a simplified illustration of the basic idea of what VADCOPs can achieve.
+
+In this scenario, it becomes feasible to execute five Keccak operations despite the limit of executing only four operations at a time.
+
+At a high level, the solution involves splitting the proof into two parts, each containing fewer rows, and then aggregating them to demonstrate the execution of five Keccak operations.
+
+![Figure: VADCOPs with Keccak](../../../img/zkEVM/ag-vadcops-with-keccak.png)
+
+The ongoing development of VADCOPs includes rewriting of both the cryptographic backend and the constraint language called _polynomial identities language_ version 2 (PIL2).
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
The ongoing development of VADCOPs includes rewriting of both the cryptographic backend and the constraint language called _Polynomial Identities Language_ version 2 (PIL2).
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+
+This scenario is of interest to us because it creates a single, unified proof that allows various batches to be verified simultaneously, thereby increasing system throughput.
+
+Two techniques of interest in this area are _proof recursion_ and _proof aggregation_.
+
+### Proof recursion
+
+Recursion, which is also called _compression_, in proof systems means the prover deals with smaller, quicker-to-verify proofs instead of large, time-consuming proofs.
+
+Essentially, the prover generates a proof at the current stage that attests to the fact that the verification of the previous stage was correctly performed.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md:143]:
```suggestion
Essentially, the prover generates a proof for the current state that attests to the fact that the verification of the previous state was correctly performed.
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+
+We now provide concrete blocks and steps used to prove correct execution of several batches by the Polygon zkEVM, using recursion and aggregation.
+
+An overview of the overall process is depicted in the figure below.
+
+![Figure: Specific recursion design](../../../img/zkEVM/ag-specific-recursion-design.png)
+
+The first proving system generates such a big proof since it has a lot of high degree polynomials. 
+
+Thereafter, the *Compression Stage* is invoked and applied on in each batch proof, aiming to reduce the number of polynomials used, and hence allowing reduction in the proof size.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md:187]: Please confirm if 'stage' is the accurate term.
If yes, the term 'invoked' might be a bit confusing. Suggest modifying to stages that are reached as part of the verification/aggregation process.

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+
+The final output of the proof recursion and aggregation process is an FFLONK proof, which is a SNARK, and it gets verified via the [ec-pairing](https://www.evm.codes/precompiled#0x08?fork=cancun) precompiled contract.
+
+## Introducing the aggregator
+
+In the Polygon zkEVM architecture, the aggregator is the component responsible for performing proof aggregation.
+
+Its role is to aggregate several proofs into one, and send the aggregated proof to the L1 Smart Contract through the $\texttt{verifyBatches()}$ function.
+
+The aggregator invokes the $\texttt{verifyBatches()}$ function on the smart contract, passing as parameters:
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
The aggregator invokes the $\texttt{verifyBatches()}$ function on the smart contract, passing the following parameters:
```

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+
+- Create a single public input, which is the hash of all the previous public keys.
+
+- Transform all other public inputs into private inputs.
+
+The figure below depicts this configuration, where public inputs are in green while private inputs are in red.
+
+![Figure: inputSNARK to Aggregator](../../../img/zkEVM/ag-inputsnark-to-aggregator-output.png)
+
+Henceforth, for the outer proof, there is a single public input in the aggregator called $\texttt{inputSnark}$, which is a Keccak hash of all the previous public inputs and outputs.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/proving-system/aggregator-proofs.md]:
```suggestion
Hence, for the outer proof, there is a single public input in the aggregator called $\texttt{inputSnark}$, which is a Keccak hash of all the previous public inputs and outputs.
```

**Review by hsutaiyu — APPROVED** [review]
Approved with suggestions.

File: docs/zkEVM/architecture/proving-system/aggregator-proofs.md (hunk 9 above)
Comment (EmpieichO) [reply docs/zkEVM/architecture/proving-system/aggregator-proofs.md:187]: Proofs go through various stages of the aggregation process. So it's stage and not state.

**Pull request merged**